src/basic/strv.c

```diff
@@ -339,21 +339,22 @@ int strv_split_extract(char ***t, const char *s, const char *separators, Extract
         return (int) n;
 }
 
-char *strv_join(char **l, const char *separator) {
+char *strv_join_prefix(char **l, const char *separator, const char *prefix) {
         char *r, *e;
         char **s;
-        size_t n, k;
+        size_t n, k, m;
 
         if (!separator)
                 separator = " ";
 
         k = strlen(separator);
+        m = prefix ? strlen(prefix) : 0;
 
         n = 0;
         STRV_FOREACH(s, l) {
                 if (s != l)
                         n += k;
-                n += strlen(*s);
+                n += strlen(*s) + m;
```

Comment:
not that it matters, but i would have swapped the operands here, after all "m" is the size of the prefix, hence it feels better if it is the first operand, and "strlen(*s)" is the size of the suffix, hence it feels better if it is the second operand...